### PR TITLE
[Polymetis] Add Cartesian impedance control

### PR DESCRIPTION
**Pull request by exhaustin**
# Description

- Add working Cartesian PD controller
- Deduplicate code across different `torchcontrol.policies` modules
- Change all ee move to applications to use Cartesian impedance controllers
- Add argument type hints that were left out in previous PR

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [x] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

If this PR introduces a change or fixes a bug, please add before and after screenshots (if this is causes a UX change) or before and after examples(this could be plain text, videos etc ) to demonstrate the change.

If this PR introduces a new feature, please also add examples or screenshots demonstrating the feature.

# Testing

Please describe the tests that you ran to verify your changes and how we can reproduce.

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [x] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [x] I have added tests that show that the PR is functional.
- [x] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I have verified that all scripts in `tests/scripts` runs on hardware.

### CODE REVIEW

File: polymetis/tests/scripts/5_continuous_grasping_hw.py [hunk 1]
```diff
@@ -54,33 +58,46 @@ def reset_policy(self):
 
         # Send PD controller
         joint_pos_current = self.arm.get_joint_angles()
-        policy = toco.policies.JointImpedanceControl(
+        policy = toco.policies.CartesianImpedanceControl(
             joint_pos_current=joint_pos_current,
-            Kp=self.arm.metadata.default_Kq,
-            Kd=self.arm.metadata.default_Kqd,
+            Kp=KP_DEFAULT,
+            Kd=KD_DEFAULT,
```
Comment (1heart) [inline polymetis/tests/scripts/5_continuous_grasping_hw.py:64]: Do we not want to grab `Kp, Kd` it from the metadata? (i.e. update the config with the values in `KP_DEFAULT`, `KD_DEFAULT`)

**Review by 1heart — APPROVED** [review]
Looks great, thank you! Let's just put a TODO somewhere to indicate which part still requires some perf work (`CartesianSpaceMoveTo`).

File: polymetis/tests/scripts/5_continuous_grasping_hw.py (hunk 1 above)
Comment (exhaustin) [reply polymetis/tests/scripts/5_continuous_grasping_hw.py]: Adding this to metadata would be a large change that would see very little use as of now.
I implemented it anyways, do you think we should keep it?

Comment (1heart) [reply polymetis/tests/scripts/5_continuous_grasping_hw.py]: Might as well do it now rather than later imo

**Comment by 1heart** [comment]
Can we verify that post-PyTorch upgrade, `CartesianSpaceMoveTo` works on hardware? And then feel free to remove the TODO comment you just added.

**Pull request merged**